### DISCUSSION TITLE
Swagger documents cannot be accessed directly through the browser after setting the proxy using root_path

### DISCUSSION THREAD

**Discussion by insistence** [fastapi/fastapi#11198] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from fastapi.staticfiles import StaticFiles
import uvicorn


app = FastAPI()

app.mount("/profile", StaticFiles(directory="static"))


@app.get('/test')
def test():
    return {'test': 'test content'}


if __name__ == '__main__':
    uvicorn.run(app='app:app', root_path='/proxy')
```

### Description

```
from fastapi import FastAPI
from fastapi.staticfiles import StaticFiles
import uvicorn


app = FastAPI()

app.mount("/profile", StaticFiles(directory="static"))


@app.get('/test')
def test():
    return {'test': 'test content'}


if __name__ == '__main__':
    uvicorn.run(app='app:app', root_path='/proxy')
```
When I run the above code, I am unable to access the Swagger document through http://127.0.0.1:8000/docs or http://127.0.0.1:8000/proxy/docs in my browser(Not Found /proxy/openapi.json). I can access the test.txt file through http://127.0.0.1:8000/profile/test.txt  in my browser.

But when I run the following code (only changing the way root_path is set), I can access the Swagger document through http://127.0.0.1:8000/docs or http://127.0.0.1:8000/proxy/docs in my browser. I am unable to access the test.txt file through http://127.0.0.1:8000/profile/test.txt  in my browser(404).
```
from fastapi import FastAPI
from fastapi.staticfiles import StaticFiles
import uvicorn


app = FastAPI(root_path='/proxy')

app.mount("/profile", StaticFiles(directory="static"))


@app.get('/test')
def test():
    return {'test': 'test content'}


if __name__ == '__main__':
    uvicorn.run(app='app:app')
```

This issue prevents me from using both Swagger documents and static files correctly at the same time.

### Operating System

Windows

### FastAPI Version

0.109.1

### Pydantic Version

2.5.3

### Python Version

3.9.18

### Uvicorn Version

0.27.0

**Comment by YuriiMotov** [comment]
I think It has been fixed in latest versions.
Let's close this for now to keep the discussions clean and organized.
If you still need help, feel free to let us know.
